roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+        # Use self._execute_module instead of self.execute_module because
+        # the latter sets self.changed.
+        result = self._execute_module('command', command_args)
+        if result.get('rc', 0) != 0 or result.get('failed'):
+            raise OpenShiftCheckException(
+                'RemoteCommandFailure',
+                'Failed to execute command on remote host: %s' % command)
+
+        return result['stdout']
+
+    def check_master(self):
+        """Gather diagnostic information on a master and ensure it can connect
+        to kubelets."""
+        if self.want_full_results:
+            conf_base_path = self.get_var('openshift.common.config_base')
+            master_conf_path = os.path.join(conf_base_path, 'master',
+                                            'master-config.yaml')
+            self.register_file('master-config.yaml', None, master_conf_path)
+
+            # XXX: Loading master logs can take in inordinate amount of
+            # time on busy systems.
+            # service_prefix = self.get_var('openshift_service_type')
+            # if self._templar is not None:
+            #     service_prefix = self._templar.template(service_prefix)
+            # try:
+            #     self.save_service_logs('%s-master-controllers' %
+            #                            service_prefix)
+            #     self.save_service_logs('%s-master-api' % service_prefix)
+            # except OpenShiftCheckException as exc:
+            #     self.register_failure(exc)
+
+        nodes = self.get_resource('nodes')
+        pods = self.get_resource('pods')
+        services = self.get_resource('services')
+
+        if self.want_full_results:
+            self.register_file('nodes.json', nodes)
+            self.register_file('pods.json', pods)
+            self.register_file('services.json', services)
+            self.register_file('endpoints.json',
+                               self.get_resource('endpoints'))
+            self.register_file('routes.json', self.get_resource('routes'))
+            self.register_file('clusternetworks.json',
+                               self.get_resource('clusternetworks'))
+            self.register_file('hostsubnets.json',
+                               self.get_resource('hostsubnets'))
+            self.register_file('netnamespaces.json',
+                               self.get_resource('netnamespaces'))
+
+            node_addrs = [(addr['address'], node)
+                          for addrs, node in [(node['status']['addresses'],
+                                               node['metadata']['name'])
+                                              for node in nodes]
+                          for addr in addrs
+                          if addr['type'] == 'InternalIP']
+            node_info = '\n'.join([' '.join(fields) for fields in node_addrs])
+            self.register_file('nodeinfo', node_info)
```

Comment:
The files in the `meta` directory of the debug.sh output (`nodeinfo`, `podinfo`, and `serviceinfo`) are only there because they contain data the script itself needs later on. You don't need to create them here, because you don't need them. (Eg, `check_node` does `for pod in self.get_resource('pods')` rather than reading back `podinfo`.)

Reply:
All right, if the files are not useful for humans, I'll delete this code.